File: packages/player/android/src/main/kotlin/br/com/suamusica/player/Media.kt
package br.com.suamusica.player

import com.google.gson.annotations.SerializedName

data class Media(
    @SerializedName("id") val id: Int,
    @SerializedName("name") val name: String,
    @SerializedName("ownerId") val ownerId: Int,
    @SerializedName("albumId") val albumId: Int,
    @SerializedName("albumTitle") val albumTitle: String,
    @SerializedName("author") val author: String,
    @SerializedName("url") val url: String,
    @SerializedName("is_local") val isLocal: Boolean,
    @SerializedName("cover_url") val coverUrl: String,
    @SerializedName("bigCover") val bigCoverUrl: String,
    @SerializedName("is_verified") val isVerified: Boolean,
    @SerializedName("shared_url") val shareUrl: String,
    @SerializedName("playlist_id") val playlistId: Int,
    @SerializedName("is_spot") val isSpot: Boolean,
    @SerializedName("isFavorite") val isFavorite: Boolean?,
    @SerializedName("fallbackUrl") val fallbackUrl: String,
    @SerializedName("indexInPlaylist") val indexInPlaylist: Int?,
    @SerializedName("catid") val categoryId: Int,
    @SerializedName("playlistTitle") val playlistTitle: String,
    @SerializedName("playlistCoverUrl") val playlistCoverUrl: String,
    @SerializedName("playlistOwnerId") val playlistOwnerId: Int
)

File: packages/player/ios/Classes/Media.swift
class Media {
    let id: Int
    let name: String
    let albumId: Int
    let albumTitle: String
    let author: String
    var url: String
    var isLocal: Bool
    let localPath: String?
    let coverUrl: String
    let bigCoverUrl: String
    let isVerified: Bool
    let playlistId: Int?
    var fallbackUrl: String?
    
    // Inicializador para a classe
    init(id: Int, name: String, albumId: Int, albumTitle: String, author: String, url: String, isLocal: Bool, localPath: String?, coverUrl: String, bigCoverUrl: String, isVerified: Bool, playlistId: Int?, fallbackUrl: String?) {
        self.id = id
        self.name = name
        self.albumId = albumId
        self.albumTitle = albumTitle
        self.author = author
        self.url = url
        self.isLocal = isLocal
        self.localPath = localPath
        self.coverUrl = coverUrl
        self.bigCoverUrl = bigCoverUrl
        self.isVerified = isVerified
        self.playlistId = playlistId
        self.fallbackUrl = fallbackUrl
    }
}
